src/interpreter/Util.h

```diff
@@ -27,82 +27,78 @@
 namespace souffle::interpreter {
 // clang-format off
 
-#define FOR_EACH_PROVENANCE(func, ...) \
-    func(Provenance, 2, __VA_ARGS__)   \
-    func(Provenance, 3, __VA_ARGS__)   \
-    func(Provenance, 4, __VA_ARGS__)   \
-    func(Provenance, 5, __VA_ARGS__)   \
-    func(Provenance, 6, __VA_ARGS__)   \
-    func(Provenance, 7, __VA_ARGS__)   \
-    func(Provenance, 8, __VA_ARGS__)   \
-    func(Provenance, 9, __VA_ARGS__)   \
-    func(Provenance, 10, __VA_ARGS__)  \
-    func(Provenance, 11, __VA_ARGS__)  \
-    func(Provenance, 12, __VA_ARGS__)  \
-    func(Provenance, 13, __VA_ARGS__)  \
-    func(Provenance, 14, __VA_ARGS__)  \
-    func(Provenance, 15, __VA_ARGS__)  \
-    func(Provenance, 16, __VA_ARGS__)  \
-    func(Provenance, 17, __VA_ARGS__)  \
-    func(Provenance, 18, __VA_ARGS__)  \
-    func(Provenance, 19, __VA_ARGS__)  \
-    func(Provenance, 20, __VA_ARGS__)  \
-    func(Provenance, 21, __VA_ARGS__)  \
-    func(Provenance, 22, __VA_ARGS__)  \
-    func(Provenance, 23, __VA_ARGS__)  \
-    func(Provenance, 24, __VA_ARGS__)  \
-    func(Provenance, 25, __VA_ARGS__)  \
-    func(Provenance, 26, __VA_ARGS__)  \
-    func(Provenance, 27, __VA_ARGS__)  \
-    func(Provenance, 28, __VA_ARGS__)  \
-    func(Provenance, 29, __VA_ARGS__)  \
-    func(Provenance, 30, __VA_ARGS__)
```

Comment:
Technically it could. I was just thinking it's a bit massive to generate provenance data structures for Btrees of arity up to size 30, for a feature that's likely almost never used. I just kept the structures for arity up to 9, which should cover 99% of the cases in practice.